src/Test.UnitTests.BinSkim.Driver/CompilerDataLoggerTests.cs

```diff
@@ -622,6 +631,139 @@ private void LogExceptionEvents(List<ExecutionException> exceptions, CompilerDat
+            string commandLineEventsId = string.Empty;
+
+            var compilerEvents = new List<EventTelemetry>();
+            var assemblyReferencesEvents = new List<EventTelemetry>();
+            var commandLineEvents = new List<EventTelemetry>();
+            var summaryEvents = new List<EventTelemetry>();
+
+            foreach (EventTelemetry telemetryEvent in telemetryGeneratedEvents)
+            {
+                IDictionary<string, string> properties = telemetryEvent.Properties;
+
+
+                if (string.IsNullOrWhiteSpace(commandLineEventsId)
+                    && properties.TryGetValue(CompilerDataLogger.CommandLineId, out string commandLineEventId))
+                {
+                    commandLineEventsId = commandLineEventId;
+                }
+
+                if (string.IsNullOrWhiteSpace(assemblyEventsId)
+                    && properties.TryGetValue(CompilerDataLogger.AssemblyReferencesId, out string assemblyEventId))
+                {
+                    assemblyEventsId = assemblyEventId;
+                }
+
+                switch (telemetryEvent.Name)
+                {
+                    case CompilerDataLogger.CompilerEventName:
+                    {
+                        compilerEvents.Add(telemetryEvent);
+
+                        if (expectedEventName == CompilerDataLogger.AssemblyReferencesEventName
+                            && !properties.ContainsKey(CompilerDataLogger.AssemblyReferencesId))
+                        {
+                            sb.AppendLine("Compiler Event is missing the `assemblyReferencesId`");
+                        }
+                        else if (expectedEventName == CompilerDataLogger.CommandLineEventName
+                            && !properties.ContainsKey(CompilerDataLogger.CommandLineId))
+                        {
+                            sb.AppendLine("Compiler Event is missing the `commandLineId`");
+                        }
+                        break;
+                    }
+                    case CompilerDataLogger.AssemblyReferencesEventName:
+                    {
+                        assemblyReferencesEvents.Add(telemetryEvent);
+
+                        if (expectedEventName == CompilerDataLogger.AssemblyReferencesEventName
+                            && !expectedChunkedContent.Contains(properties[CompilerDataLogger.ChunkedAssemblyReferences]))
+                        {
+                            sb.AppendLine(
+                                $"Unexpected `{CompilerDataLogger.AssemblyReferencesEventName}` chunked content: " +
+                                $"`{properties[CompilerDataLogger.ChunkedAssemblyReferences]}` " +
+                                $"expected: `{expectedChunkedContent}`");
+                        }
+
+                        string currentAssemblyEventId = properties["assemblyReferencesId"];
+
+                        if (assemblyEventsId != properties["assemblyReferencesId"])
```

Comment:
>if (assemblyEventsId != properties["assemblyReferencesId"]) [](http://example.com/codeflow?start=24&length=59)

defined currentAssemblyEventId right before this line,
were you planning to use it here:
if (assemblyEventsId != currentAssemblyEventId)

Reply:
Did the same for the command line event too.